synthetics/datadog/datadog.go

```diff
@@ -0,0 +1,201 @@
+package synthetics
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/zorkian/go-datadog-api"
+	"go.uber.org/zap"
+)
+
+// TestManager synchronize synthetics tests in Datadog witm existing Kubernetes Ingress Endpoints
+type TestManager struct {
+	Client              Client
+	Logger              *zap.Logger
+	AlertMessage        string
+	CheckInterval       int
+	DefaultTag          string
+	Tags                []string
+	AdditionalEndpoints []string
+	Enabled             bool
+}
+
+// Client is an interface that clients implement to manage synthetic tests in Datadog.
+type Client interface {
+	CreateSyntheticsTest(syntheticsTest *datadog.SyntheticsTest) (*datadog.SyntheticsTest, error)
+	GetSyntheticsTests() ([]datadog.SyntheticsTest, error)
+	DeleteSyntheticsTests(publicIds []string) error
+}
+
+// NewTestManager creates a new datadog.Client
+func NewTestManager(apiKey string, appKey string) (*TestManager, error) {
+	if apiKey == "" {
+		return &TestManager{}, errors.New("datadog api key is required")
+	}
+	if appKey == "" {
+		return &TestManager{}, errors.New("datadog application key is required")
+	}
+	return &TestManager{
+		Client: datadog.NewClient(apiKey, appKey),
+	}, nil
+}
+
+// Contains return whether a slice contains a specific value
+func Contains(slice []string, val string) bool {
+	for _, n := range slice {
+		if val == n {
+			return true
+		}
+	}
+	return false
+}
+
+// ContainsColon return whether a slice contains a specific value before the first colon
+func ContainsColon(slice []string, val string) bool {
+	for _, n := range slice {
+		if strings.ContainsAny(n, ":") {
+			s := strings.Split(n, ":")
+			n = s[0]
+		}
+		if val == n {
+			return true
+		}
+	}
+	return false
+}
+
+// createManagedSyntheticsTest configures and create a new synthetics test in Datadog
+func (tm *TestManager) createManagedSyntheticsTest(endpoint string, port int) (*datadog.SyntheticsTest, error) {
+	newOptions := &datadog.SyntheticsOptions{}
+	newOptions.SetAcceptSelfSigned(false)
+	newOptions.SetTickEvery(tm.CheckInterval)
+
+	expiryAssertion := &datadog.SyntheticsAssertion{}
+	expiryAssertion.SetType("certificate")
+	expiryAssertion.SetOperator("isInMoreThan")
+	expiryAssertion.Target = 12
+
+	newRequest := &datadog.SyntheticsRequest{}
+	newRequest.SetHost(endpoint)
+	newRequest.SetPort(port)
+
+	newConfig := &datadog.SyntheticsConfig{}
+	newConfig.Assertions = []datadog.SyntheticsAssertion{*expiryAssertion}
+	newConfig.SetRequest(*newRequest)
+
+	tags := tm.Tags
+	tags = append(tags, tm.DefaultTag)
+
+	newTest := &datadog.SyntheticsTest{Locations: []string{"aws:ap-northeast-1"}, Tags: tags}
```

Comment:
Can you please make this an ENV var? e.g. `[]string "SYNTHETICS_DEFAULT_LOCATIONS"`. We can think about making this configurable per-endpoint later.

```suggestion
	newTest := &datadog.SyntheticsTest{Locations: []string{"aws:ap-northeast-1"}, Tags: tags}
```

Reply:
Added in https://github.com/lainra/certificate-expiry-monitor-controller/pull/1/commits/e9678afffcfd76eab988b50fcef1af3d7fad6464 and https://github.com/lainra/certificate-expiry-monitor-controller/pull/1/commits/5be1f70f74cbd456b94d42960fb99101317f81a0